instance/tasks.py

```diff
@@ -339,3 +344,111 @@ def cleanup_old_betatest_users():
         users_to_delete = queryset.filter(is_active=False, profile__modified__lte=delete_cutoff)
         logger.info('Deleting %s users.', users_to_delete.count())
         users_to_delete.delete()
+
+
+class KillZombiesRunner:
+    """
+    Helper class to run `kill_zombies_periodically`
+    """
+
+    def __init__(self):
+        self.region: str = getattr(settings, "OPENSTACK_REGION", "gra")
+        self.recipient: list = [email for _, email in getattr(settings, "ADMINS", [])]
+        self.threshold: int = getattr(settings, "KILL_ZOMBIES_WARNING_THRESHOLD")
+
+    def send_email(self, subject: str, body: str) -> int:
+        """
+        Utility method for sending emails
+        """
+        if self.recipient:
+            email = EmailMessage(
+                subject, body, settings.DEFAULT_FROM_EMAIL, self.recipient
+            )
+            return email.send()
+        else:
+            logging.info("Email recipient is undefined. Skipping email.")
+            return None
+
+    def trigger_warning(self):
+        """
+        Warns when more than `threshold` VMs will be terminated
+        """
+        subject = "{} Zombie instances over current threshold ({})".format(
+            datetime.now().strftime("%Y-%m-d"), self.threshold
+        )
+        body = (
+            "The number of instances in region {} to be terminated"
+            "is over KILL_ZOMBIES_WARNING_THRESHOLD ({}).\n"
+        ).format(self.region, self.threshold)
+        logging.info(
+            "%s\nSending an email notification to: %s",
+            body, self.recipient
+        )
+        self.send_email(subject, body)
+
+    def handle_dry_run_output(self, stdout: str) -> int:
+        """
+        If there are servers to terminate, a `kill_zombies` dry run will warn:
+        "Would have terminated {} zombies" to stdout.
+        """
+        if "No servers found in region" in stdout:
+            return 0
+        try:
+            return int(stdout.split("Would have terminated")[1].split()[0])
+        except IndexError:
```

Comment:
@spokerman12, what if the `int()` call returned a `ValueError`?

Reply:
This was modified following previous suggestions.

Reply:
@spokerman12, I don't see any code changes handling the `ValueError` that might be thrown by the `int()` call now?

Reply:
This is what happens if that value is not an integer.

```
 File "/var/www/.pyenv/versions/opencraft/lib/python3.6/site-packages/huey/api.py", line 342, in _execute
    task_value = task.execute()
  File "/var/www/.pyenv/versions/opencraft/lib/python3.6/site-packages/huey/api.py", line 690, in execute
    return func(*args, **kwargs)
  File "/var/www/.pyenv/versions/opencraft/lib/python3.6/site-packages/huey/contrib/djhuey/__init__.py", line 128, in inner
    return fn(*args, **kwargs)
  File "/var/www/opencraft/instance/tasks.py", line 449, in kill_zombies_task
    kill_zombies_periodically()
  File "/var/www/opencraft/instance/tasks.py", line 460, in kill_zombies_periodically
    task_runner.run()
  File "/var/www/opencraft/instance/tasks.py", line 413, in run
    num_of_zombies = self.get_zombie_servers_count(dry_run_output)
  File "/var/www/opencraft/instance/tasks.py", line 397, in get_zombie_servers_count
    num_instances = int(stdout.split("Would have terminated")[1].split()[0])
ValueError: invalid literal for int() with base 10: 'abc'
```

Reply:
`stdout` is well-defined here https://github.com/open-craft/opencraft/blob/f70334b0b80e12a0f8ddbd0ee981171479774eb5/instance/management/commands/kill_zombies.py#L85 as a well-formed string or None. If those strings were badly-formatted, that's an issue of not having tests for `kill_zombies` in the first place.
Added.